File: output/ifc4x3_dev_73740fe4/entity/IfcBSplineCurve.java
package com.topoplanet.ifc4x3_dev_73740fe4.entity;
import com.topoplanet.test.type.*;


import java.util.List;

public abstract class IfcBSplineCurve extends IfcBoundedCurve {

    public IfcInteger Degree;
    public IfcBSplineCurveForm CurveForm;
    public IfcLogical ClosedCurve;
    public IfcLogical SelfIntersect;
    // INVERSE attributes:
    // UpperIndexOnControlPoints : IfcInteger := (SIZEOF(ControlPointsList) - 1);
    // ControlPoints : ARRAY [0:UpperIndexOnControlPoints] OF IfcCartesianPoint := IfcListToArray(ControlPointsList,0,UpperIndexOnControlPoints);
    // INVERSE attributes:
    // SameDim : SIZEOF(QUERY(Temp <* ControlPointsList |
    // Temp.Dim <> ControlPointsList[1].Dim))
    // = 0;

    // SUPERTYPE OF:
    // - IfcBSplineCurveWithKnots SUBTYPE OF IfcBoundedCurve

    // WHERE constraints:
    // SameDim : SIZEOF(QUERY(Temp <* ControlPointsList |
    // Temp.Dim <> ControlPointsList[1].Dim))
    // = 0;
}

File: src/main/resources/output/IFC4X3_csharp/Entities/IfcRepresentationItem/IfcGeometricRepresentationItem/IfcCurve/IfcBoundedCurve/IfcBSplineCurve/IfcBSplineCurve.cs
public class IfcBSplineCurve : IfcBoundedCurve
{
    public IfcInteger Degree { get; set; }
    public List<IfcCartesianPoint> ControlPointsList { get; set; }
    public IfcBSplineCurveForm CurveForm { get; set; }
    public IfcLogical ClosedCurve { get; set; }
    public IfcLogical SelfIntersect { get; set; }

    // === EXTENDED BY ===
    // IfcBSplineCurveWithKnots

    // === WHERE CLAUSES ===
    // SameDim : SIZEOF(QUERY(Temp <* ControlPointsList | Temp.Dim <> ControlPointsList[1].Dim)) = 0

    // === DERIVE CLAUSES ===
    // UpperIndexOnControlPoints : IfcInteger := (SIZEOF(ControlPointsList) - 1)
    // ControlPoints : ARRAY [0:UpperIndexOnControlPoints] OF IfcCartesianPoint := IfcListToArray(ControlPointsList,0,UpperIndexOnControlPoints)
}
